internal/providers/azure/pke/workflow/create_cluster.go

```diff
@@ -162,22 +178,28 @@ func CreateClusterWorkflow(ctx workflow.Context, input CreateClusterWorkflowInpu
 
 	err = workflow.ExecuteChildWorkflow(ctx, cluster.RunPostHooksWorkflowName, postHookWorkflowInput).Get(ctx, nil)
 	if err != nil {
-		_ = setClusterErrorStatus(ctx, input.ClusterID, err)
 		return err
 	}
 
 	return nil
 }
 
+type decodableError struct {
+	Message string
+}
+
+func (d decodableError) Error() string {
+	return d.Message
+}
+
 func waitForMasterReadySignal(ctx workflow.Context, timeout time.Duration) error {
-	signalName := "master-ready"
 	signalChan := workflow.GetSignalChannel(ctx, signalName)
 	signalTimeoutTimer := workflow.NewTimer(ctx, timeout)
 	signalTimeout := false
 
+	var signalValue decodableError
 	signalSelector := workflow.NewSelector(ctx).AddReceive(signalChan, func(c workflow.Channel, more bool) {
-		c.Receive(ctx, nil)
-		workflow.GetLogger(ctx).Info("Received signal!", zap.String("signal", signalName))
+		c.Receive(ctx, &signalValue)
```

Comment:
I guess this is assumes that a successful event will not send anything in the signal. I think it would probably be better to explicitly send a separate signal on error.

Then each signal (timeout and bootstrap error) could instead set an error directly.

Reply:
yes, the payload of the node ready signal is the error. err=nil is quite common pattern to signal success in go

Reply:
Except this isn't an error and it's not returned from a function, but unmarshaled from an encoded something, making the behavior implicit.